### PR TITLE
fix: put, copy, completeMPU should reject verId

### PR DESCRIPTION
**Pull request by electrachong**

### CODE REVIEW

File: lib/api/apiUtils/object/checkQueryVersionId.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { errors } from 'arsenal';
```
Comment (LaurenSpiegel) [inline lib/api/apiUtils/object/checkQueryVersionId.js]: can we put this file into the versioning.js that is in apiUtils/object already?

File: tests/functional/aws-node-sdk/lib/utility/customS3Request.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+export default function customS3Request(action, params, buildParams, callback) {
+    const method = action.bind(s3);
+    const request = method(params);
+    const { headers, query } = buildParams;
+    // modify underlying http request object created by aws sdk
+    request.on('build', () => {
+        Object.assign(request.httpRequest.headers, headers);
+        if (query) {
+            const qs = querystring.stringify(query);
+            // NOTE: that this relies on there not being on query string in the
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/lib/utility/customS3Request.js]: on s/b a

File: lib/api/apiUtils/object/checkQueryVersionId.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * Checks for versionId in request query and returns error if it is there
+ * @param {object} query - request query
+ * @return {(Error|undefined)} - customized InvalidArgument error or undefined
+ */
+export default function checkQueryVersionId(query) {
+    if (query && query.versionId !== undefined) {
+        const customMsg = 'This operation does not accept a version-id.';
```
Comment (LaurenSpiegel) [inline lib/api/apiUtils/object/checkQueryVersionId.js]: Is this the AWS specific error message?

Comment (electrachong) [inline lib/api/apiUtils/object/checkQueryVersionId.js]: Yes

File: tests/functional/aws-node-sdk/lib/utility/customS3Request.js (hunk 2 above)
Comment (electrachong) [inline tests/functional/aws-node-sdk/lib/utility/customS3Request.js]: What does this mean?

File: lib/api/apiUtils/object/checkQueryVersionId.js (hunk 1 above)
Comment (electrachong) [inline lib/api/apiUtils/object/checkQueryVersionId.js]: updated

**Comment by electrachong** [comment]
@ironman-machine

File: tests/functional/aws-node-sdk/lib/utility/customS3Request.js (hunk 2 above)
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/lib/utility/customS3Request.js]: "on" should be "a".  
"this relies on there not being **a** query..."

Comment (electrachong) [inline tests/functional/aws-node-sdk/lib/utility/customS3Request.js]: Oh, got it

**Comment by LaurenSpiegel** [comment]
@ironman-machine try

**Comment by electrachong** [comment]
@ironman-machine try

**Comment by electrachong** [comment]
end to end passed: http://ci.ironmann.io/gh/scality/Integration/12268

**Pull request merged**